### PR TITLE
Improve and modernize the submodule cheat sheet

### PR DESCRIPTION
**Pull request by phil-blain**
## Overview
**TL;DR**
The "Submodules vs. Subtrees" cheat sheet was missing some important information about working with projects that have submodules, mainly that `git pull` and `git checkout` do not, by default, update the working directory of the submodule(s), causing much confusion and sorrow.

I also improved the other sections of this cheat sheet to add details and use more modern Git commands and flags. Each commit message explain the changes in details.

<Link to related issue. Type `closes #RELATEDISSUENUMBER` to establish a link.>

### Questions
The `submodule.recurse` config is super useful, but it works correctly for `git pull` only starting with Git 2.15. Should that be mentioned ?

### CODE REVIEW

**Comment by brianamarie** [comment]
Thank you for this contribution, @phil-blain! cc @a-a-ron for comms duty 👀

**Comment by a-a-ron** [comment]
Hey @phil-blain, thanks for the updates! I reviewed about half the changes and then needed to direct my attention elsewhere. I can review the rest tomorrow morning.

File: downloads/submodule-vs-subtree-cheat-sheet.md [hunk 1]
```diff
@@ -216,13 +317,18 @@ Always show the submodule log when you diff:
...
+Show a short summary of submodule changes in your `git status` message:
+
+    git config --global status.submoduleSummary true
+
+Make `push` default to `--recurse-submodules=on-demand`:
+
+    git config --global push.recurseSubmodules on-demand
 
-    git config status.submoduleSummary true
+Make all commands (except `clone`) default to `--recurse-submodules` if they support the flag:
```
Comment (a-a-ron) [inline downloads/submodule-vs-subtree-cheat-sheet.md]:
```suggestion
Make all commands (except `clone`) default to `--recurse-submodules` if they support the flag (this works for `git pull` since Git 2.15):
```

**Comment by a-a-ron** [comment]
> The submodule.recurse config is super useful, but it works correctly for git pull only starting with Git 2.15. Should that be mentioned ?

Hey @phil-blain, I made one suggested commit based on this comment. Yes, I think it should be listed. Thanks for adding all of this information!

**Comment by phil-blain** [comment]
@a-a-ron I just committed your suggestion. Thanks!

**Comment by a-a-ron** [comment]
Thanks @phil-blain. These updates are very welcomed!

**Comment by phil-blain** [comment]
Glad I could help! heads-up: I plan to link to the training-kit home page from the [git-scm.com documentation page](git-scm.com/docs) (see git/git-scm.com#1425) At the moment it links directly to the English PDF cheat sheet.

**Comment by a-a-ron** [comment]
> I plan to link to the training-kit home page from the git-scm.com documentation page (see git/git-scm.com#1425) At the moment it links directly to the English PDF cheat sheet.

This is great. I was unaware that it was only linked to the PDF. Thank you for opening up that issue!

**Comment by brianamarie** [comment]
Thank you all! ✨
